docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,132 @@
+## Backpack-android architecture overview:
+
+Backpack-android is a Android a traditional gradle Android lib project, composed of two modules.
+
+- app: The example app
+- Backpack: The Backpack Android components
+
+## App
+
+### Stories
+
+Stories are defined programmatically in tree like structure that is then used to build the UI.
+
+Important classes/packages are: 
+
+- **`{javaSrc}/data/ComponentRegistry`**
+
+  Defines a tree of `stories` used to build the UI:
+
+  ```kotlin
+  "Chip" story NodeData { ChipStory() },
+  "Panel" story NodeData { Story of R.layout.fragment_panel },
+  "Spinner" story NodeData({ children -> SubStory of children },
+    mapOf(
+      "Default" story NodeData { Story of R.layout.fragment_spinner },
+      "Small" story NodeData { Story of R.layout.fragment_spinner_small }
+    )),
+  ```
+
+- **`res/layout/`**
+  Where layouts for each story is defined. They are named `fragment_{story_name}`
+
+- **`{javaSrc}/stories`**
+  If a story needs dynamic behaviours it will have an accompanying Kotlin class inside the `stories` package.
+
+### Theming
+
+Theming for the example app is set in the `Settings` view and saved in the local shared preferences 
+file. 
+
+Important classes/packages are:
+
+- **`{javaSrc}/SettingsActivity`**
+  Activity that shows the theming configuration.
+
+- **`{javaSrc}/ThemeApplier`**
+  Implementation of a `onActivityCreated` callback that applies the current theme to the activity.
+
+- **`{javaSrc}/BackpackDemoApplication`**
+  Where all initialization logic resides, including registering the current theme.
+
+- **`{javaSrc}/data/SharedPreferences`**
+  Used to read/write shared preferences for the example app, including the current theme.
+
+
+## Backpack
+
+The backpack lib is where all components live. All packages, except `utils` are Backpack components.
+
+### Component's architecture
+
+There are only a few things that should be observed about the architecture of the components: 
+
+1. Whenever possible the Backpack component should extend from the Android base component and add the additional extra functionality instead of wrapping the Android component inside another `ViewGroup`. This is to keep the view hierarchy as small as possible.
+2. All props should be available via XML and programmatically. Whenever a component is available in both XML and code all its properties should also be available in both. 
+3. Changing a property programmatically should immediately reflect on the UI.
+4. Components should be "extensible", i.e. `open class` in Kotlin. It's also important to think about the component being used that way when writing/reviewing it.
+
+More can be found here: https://github.com/Skyscanner/backpack-android/tree/master/decisions
```

Comment:
Maybe this should live in it's own `### decisions` section?

Reply:
That's a good point. Can we leave it here for now and I can move it do decision files in the future?